### PR TITLE
tfxunit2junit: handle package installation, improve error handling

### PR DESCRIPTION
**Pull request by thrix**
To improve the experience for the user, this merge request,
in case of failed package installation and preparation playbooks.
It adds relevant error logs for the user as a failed testcase.
If the installation passed, no logs are added, as in this case
this information is for the user irrelevant (he should not care)

Testcases parsing is moved to a function for readability.

The check for no testcases is moved at the end, in case no testcases
were added we add the generic infra error.

Tmt erros are classified as failures.

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by thrix** [comment]
PTAL @msrb if you have time. I guess it would be a good idea to add some tests for all corner cases. Do you want that in this PR or we can do next?

File: scripts/tfxunit2junit.py [hunk 1]
```diff
@@ -123,6 +146,44 @@ def load_tf_xunit(tf_xunit_path):
         tf_xunit = f.read().replace('\\"', '"')
     return tf_xunit
 
+def get_artifact_installation_logs(url):
```
Comment (thrix) [inline scripts/tfxunit2junit.py:149]:
```suggestion

def get_artifact_installation_logs(url):
```

File: scripts/tfxunit2junit.py [hunk 2]
```diff
@@ -123,6 +146,44 @@ def load_tf_xunit(tf_xunit_path):
...
+    outputs = []
+
+    for log in logs:
+        logname = log.attrib['href']
+        outputs.append('Stage: {name}\n\n{log}'.format(
+            name=logname,
+            log=get_test_logs(os.path.join(url, logname)).decode()
+        ))
+
+    return outputs
```
Comment (thrix) [inline scripts/tfxunit2junit.py:186]:
```suggestion
    return outputs

```

File: scripts/tfxunit2junit.py [hunk 3]
```diff
@@ -173,59 +346,37 @@ def main(args):
...
+            # let's fabricate at least this dummy "infrastructure" test so we can show URLs to docs and issue tracker
+            add_error(
+                output_xml, 'infrastructure', 'No tests were run.',
+                'Error: tests failed to run. This is likely an infrastructure issue.',
+                issues_url=args.issues_url
+            )
+            output_xml.attrib['errors'] = '1'
 
     objectify.deannotate(root_output_xml, cleanup_namespaces=True, xsi_nil=True)
     print(etree.tostring(root_output_xml, pretty_print=True).decode())
```
Comment (thrix) [inline scripts/tfxunit2junit.py:375]:
```suggestion
    print(etree.tostring(root_output_xml, pretty_print=True).decode())

```

**Comment by msrb** [comment]
>I guess it would be a good idea to add some tests for all corner cases. Do you want that in this PR or we can do next?

I'd say don't bother with tests right now. I haven't laid down any foundation for testing of this script, i.e. no test script/framework, so it would be unreasonalble to ask you to do what I should have done long time ago ;)

**Comment by thrix** [comment]
@msrb anyway :) let's try it! at least few examples. Trying to bake something

File: scripts/tfxunit2junit.py [hunk 4]
```diff
@@ -173,59 +344,37 @@ def main(args):
...
+        try:
+            output_xml.attrib['name'] = testsuite.attrib['name']
+        except KeyError:
+            output_xml.attrib['name'] = 'testsuite {}'.format(count)
+
+        output_xml = parse_package_installation(testsuite, output_xml)
+
+        output_xml = parse_testcases(testsuite, output_xml)
+
+        # not testcases added, this is a real infra error :(
```
Comment (thrix) [inline scripts/tfxunit2junit.py]:
```suggestion
```

**Comment by thrix** [comment]
tmt run now works, but the tests fail, how weird

**Comment by msrb** [comment]
This is going to be a huge improvement -- people will love it ;) Thank you for the pull request, LGTM.

**Comment by msrb** [comment]
Just for the record, I tested the changes here: https://osci-jenkins-1.ci.fedoraproject.org/job/fedora-ci/job/dist-git-pipeline/view/change-requests/job/PR-32/
